### PR TITLE
Add LLM running code and docs

### PR DESCRIPTION
**Pull request by evgenyrp**
I'm adding this experimental but mostly reproducible code to a separate folder so that it's not lost when we want to continue experimenting with LLMs or someone else wants to try it.

Also, maybe I missed something else that we can try. Please look at it on a high level and let me know what you think.

I produced 1M and 10M parallel sentences datasets with Gemma 27b and greedy decoding and used them for fine-tuning, which led to some improvement in the COMET scores for en-ru (student pre-trained on the best 50M NLLB).  Of course not as significant as with NewsPALM for en-de that I also tried (NewsPALM was produced with significantly more compute: bigger model and using 512 candidates for MBR and QE reranking). I'm not reporting the full results of the fine-tuning experiments here as they are not finalized.

Related to #1037 #994 #767 

[skip ci]

### CODE REVIEW

File: experiments/llmaat/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+This work follows the paper [Introducing the NewsPaLM MBR and QE Dataset:
+LLM-Generated High-Quality Parallel Data Outperforms Traditional
+Web-Crawled Data](https://arxiv.org/pdf/2408.06537).
+
+It also uses the evaluation dataset and the prompt from [WMT24++: Expanding the Language Coverage of WMT24 to 55 Languages & Dialects](https://arxiv.org/html/2502.12404v1).
+
+
+## Selecting a corpus
+
+The idea is to have a diverse monolingual dataset to translate by an LLM.
```
Comment (ZJaume) [inline experiments/llmaat/README.md:16]: Is there a reason why this is only focused on monolingual data? If the goal is to use LLM as a teacher, I would generate distilled data at least from all parallel corpora available (or sample big corpora), then add monolingual data. Parallel data could cover certain domain areas that are not well covered by monolingual data? Also it has the advantage that if the LLM inference produces an n-best list (regardless of whether it is greedy decoding or beam decoding), the hypotheses can be filtered by chrF score like we are doing right now with distilled data (potentially removing some hallucinations).

Comment (evgenyrp) [reply experiments/llmaat/README.md:16]: For sure, we could go with the same path we currently use for teacher distillation. The idea here was to pre-train on super clean existing parallel sentences and augment with synthetic high-quality examples. We can go with purely LLM-generated examples, similar to how we do distillation now and then we can do n-best reranking in a similar way. The limitation is inference costs. If we pretrain on the OPUS corpus, we need to generate a lot smaller fine-tuning dataset with an LLM. For example, 50M OPUS + 10M synthetic. Otherwise, we would need to generate 60M examples. We can definitely try that too.

File: experiments/llmaat/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# LLM as a teacher
```
Comment (gregtatum) [inline experiments/llmaat/README.md]:
```suggestion
# LLM as a teacher (llmaat)
```

Comment (gregtatum) [reply experiments/llmaat/README.md]: I found the `llmaat` acronym confusing.
